buzzmoon-frontend/buzzmoon-vite/src/Components/BackendActor/backend-actor.js

```diff
@@ -65,18 +62,95 @@ export default class BackendActor {
     );
 
     const isCorrect = this.checkAnswerCorrectness(givenAnswer, data.answers);
-    return {question: data, celerity: celerity, points:(isCorrect ? this.pointsFromCelerity(celerity) : 0),
-      givenAnswer: givenAnswer, isCorrect:isCorrect};
+    return {
+      question: data,
+      celerity: celerity,
+      points:(isCorrect ? this.pointsFromCelerity(celerity) : 0),
+      givenAnswer: givenAnswer,
+      isCorrect:isCorrect
+    };
   }
 
-  static checkAnswerCorrectness(givenAnswer, acceptableAnswers){
+  // checks the correctness of a given answer by comparing to a list of acceptable answers
+  static checkAnswerCorrectness(givenAnswer, acceptableAnswers) {
     let answerChecks = acceptableAnswers.map((ans) => (ans.toLowerCase() == givenAnswer.toLowerCase()));
     return answerChecks.reduce((acc, elem) => (acc || elem), false);
   }
 
+  // determines how many points a user earns by the percentage of the question that was read at the time they buzzed
   static pointsFromCelerity(celerity) {
     return 10 + Math.floor(Math.sqrt(celerity) / 0.1);
   }
 
-}
+  // prepares an object to be the argument for a POST request to the server
+  // the object represents contains all the metadata, question anwers, and audio to create the game on the backend
+  static prepareGameData(title, description, questions) {
+    return {
+      title: title.trim(),
+      description: description.trim(),
+      questionCount: questions.length,
+      questions:
+        (questions.map((q) => ({
+          questionNumber: q.number,
+          answers: q.answers.split(',').map((s) => (s.trim())),
+          audioFile: q.audioFile,
+        }))),
+    };
+  }
+
+  // duplicates the validation of game data that occurs on the backend, to prevent unnecessary bandwidth usage
+  static validateGameData(gameData) {
+    if (!gameData.title || gameData.title.trim() === '') {
+      throw 'No title provided';
+    }
+
+    if (!gameData.description || gameData.description.trim() === '') {
+      throw 'No description provided';
+    }
+
+    if (!gameData.questionCount || gameData.questionCount === 0) {
+      throw "Can't create a game with zero questions";
+    }
 
+    if (!gameData.questions || gameData.questions.length === 0) {
+      throw "Can't create a game with zero questions";
+    }
+
+    gameData.questions.map((q) => {
+      if (!q.answers || q.answers.length === 0 || (q.answers.length === 1 && q.answers[0] === '')) {
```

Comment:
is there a cleaner way to check if the input is empty? 
for `q.answers.length === 1 && q.answers[0] === ''`